pkg/logs/logs.go

```diff
@@ -0,0 +1,127 @@
+package logs
+
+import (
+	"io"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/redhat-developer/odo/pkg/kclient"
+	odolabels "github.com/redhat-developer/odo/pkg/labels"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type LogsClient struct {
+	kubernetesClient kclient.ClientInterface
+}
+
+func NewLogsClient(kubernetesClient kclient.ClientInterface) *LogsClient {
+	return &LogsClient{
+		kubernetesClient: kubernetesClient,
+	}
+}
+
+var _ Client = (*LogsClient)(nil)
+
+func (o *LogsClient) DevModeLogs(componentName string, namespace string) (map[string]io.ReadCloser, error) {
```

Comment:
Indeed. But I'm separating things into `DevModeLogs`, `DeployModeLogs` and `AllModeLogs`. This is because we are going to support flags that support showing logs for different modes. You don't see the other two methods because this PR is targeting only dev mode.